x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -326,7 +326,9 @@ public void resolveAsMergedMapping(
         FieldCapabilitiesRequest fieldRequest = createFieldCapsRequest(indexWildcard, fieldNames, indicesOptions, runtimeMappings);
         client.fieldCaps(
             fieldRequest,
-            listener.delegateFailureAndWrap((l, response) -> l.onResponse(mergedMappings(typeRegistry, indexWildcard, response)))
+            listener.delegateFailureAndWrap(
+                (l, response) -> l.onResponse(mergedMappings(typeRegistry, indexWildcard, response, (s, cap) -> null))
```

Comment:
Now that the PR is in place, you can override the method to allow the old method signature (without a validity specified) to minimize the amount of noise.

Reply:
I had mixed feelings about this approach, as I find it harder to determine what are the method's true capabilities. Plus `IndexResolver` is so cluttered with lots of methods that do slightly different things in slightly different scenarios and have the same name.

Reply:
I'll leave it as is. Those class' methods are confusing enough.